packages/node_modules/@ciscospark/widget-recents/src/selector.js

```diff
@@ -211,6 +211,18 @@ const getRecentSpacesWithDetail = createSelector(
   }
 );
 
+const getCurrentUserWithAvatar = createSelector(
+  [getCurrentUser, getAvatars], (currentUser, avatars) => {
+    let user;
+
+    if (currentUser && currentUser.id) {
+      user = Object.assign(currentUser, {img: avatars.getIn(['items', currentUser.id])});
```

Comment:
This is doing the same thing as adding the properties to the object, you want to do `Object.assign({}, currentUser, {img` so that the "[target](http://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign)" is a new js object.

Reply:
I don't think thats what it does. it merges the new object `{img...` with `currentUser` and returns a new object which is what I'm passing.

Reply:
Adam, I cant make it work on my machine. your suggested statement does not reflect the currentUser properties into a new object. you are more than welcome to make the changes.

Reply:
I'm keeping it as is.

Reply:
Just tried it out locally, that is because the current user object is a immutable record. This works as expected:

```js
user = Object.assign({}, currentUser.toJS(), {img: avatars.getIn(['items', currentUser.id])});
```